Models/Core/ApsimFile/Converter.cs

```diff
@@ -1383,6 +1384,48 @@ private static void UpgradeToVersion70(JObject root, string fileName)
             }
         }
 
+        /// <summary>
+        /// Alters all existing linint functions to have a child variable reference IFunction called XValue instead of a
+        /// string property called XProperty that IFunction then had to locate
+        /// </summary>
+        /// <param name="root">The root JSON token.</param>
+        /// <param name="fileName">The name of the apsimx file.</param>
+        private static void UpgradeToVersion71(JObject root, string fileName)
+        {
+            foreach (JObject linint in JsonUtilities.ChildrenRecursively(root, "LinearInterpolationFunction"))
+            {
+                VariableReference varRef = new VariableReference();
+                varRef.Name = "XValue";
+                varRef.VariableName = linint["XProperty"].ToString();
+                JsonUtilities.AddModel(linint, varRef);
+                linint.Remove("XProperty");
+            }
+        }
+
+        /// <summary>
+        /// Remove .Value() from all variable references because it is redundant
+        /// </summary>
+        /// <param name="root">The root JSON token.</param>
+        /// <param name="fileName">The name of the apsimx file.</param>
+        private static void UpgradeToVersion72(JObject root, string fileName)
+        {
+            foreach (var varRef in JsonUtilities.ChildrenRecursively(root, "VariableReference"))
+                varRef["VariableName"] = varRef["VariableName"].ToString().Replace(".Value()", "");
+
+            foreach (var report in JsonUtilities.ChildrenOfType(root, "Report"))
+                JsonUtilities.SearchReplaceReportVariableNames(report, ".Value()", "");
+
+            foreach (var graph in JsonUtilities.ChildrenOfType(root, "Series"))
+            {
+                graph["XFieldName"] = graph["XFieldName"].ToString().Replace(".Value()", "");
```

Comment:
I think this might be the problem. There are probably some graphs out there (e.g. in standard toolbox) whose XFieldName is set to null. You could try something like this:

```suggestion
                if (graph["XFieldName"] != null)
                    graph["XFieldName"] = graph["XFieldName"].ToString().Replace(".Value()", "");
```

You'd probably want to add the same check for the y field name as well.